Tasks/Common/azure-arm-rest/azure-arm-mysql.ts

```diff
@@ -0,0 +1,320 @@
+            });
+
+        this.client.beginRequest(httpRequest).then(async (response) => {
+            var deferred = Q.defer<azureServiceClient.ApiResult>();
+            var statusCode = response.statusCode;
+            tl.debug("Response for delete firewall rule " +statusCode);
+            if (statusCode != 200 && statusCode != 202 && statusCode !=204) {
+                // Generate Error
+                deferred.reject(new azureServiceClient.ApiResult(azureServiceClient.ToError(response)));
+            }
+            else {
+                // Generate Response
+                deferred.resolve(new azureServiceClient.ApiResult(null, response.body));
+            }
+            return deferred.promise;
+        }).then((apiResult: azureServiceClient.ApiResult) => callback(apiResult.error, apiResult.result),
+            (error) => callback(error));
+    }
+
+    /**
+     * Get firewall rule of mysql server
+     * @param resourceGroupName     resource group name of mysql server 
+     * @param serverName            mysql server name
+     * @param firewallRuleName      firewall rule name to be deleted 
+     * @param callback              response callback 
+     */
+    public get(resourceGroupName: string, serverName: string, firewallRuleName: string, callback?:  azureServiceClient.ApiCallback) {
+        var client = this.client;
+        if (!callback) {
+            throw new Error(tl.loc("CallbackCannotBeNull"));
+        }
+
+        // Validate
+        try {
+            this.client.isValidResourceGroupName(resourceGroupName);
+            if(!this.isNameValid(serverName)){
+                throw new Error(tl.loc("MysqlServerNameCannotBeEmpty"));
+            }
+            if(!this.isNameValid(firewallRuleName)){
+                throw new Error(tl.loc("FirewallRuleNameCannotBeNull"));
+            }
+        }
+        catch (error) {
+            return callback(error);
+        }
+
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'GET';
+        httpRequest.headers = {};
+        httpRequest.uri = this.client.getRequestUri(
+            '//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/firewallRules/{firewallRuleName}',
+            {
+                '{resourceGroupName}': resourceGroupName,
+                '{serverName}': serverName,
+                '{firewallRuleName}': firewallRuleName
+            });
+        tl.debug("Calling get firewall ");
+        this.client.beginRequest(httpRequest).then(async (response) => {
+            var deferred = Q.defer<azureServiceClient.ApiResult>();
+            var statusCode = response.statusCode;
+            tl.debug("Response for get firewall rule " +statusCode);
+            if (statusCode === 200) {
+                // Generate Response
+                tl.debug("Response for get firewall rule " + JSON.stringify(response));
```

Comment:
Can we print the whole response, before if condition?
It might provide some more info for failure case.

Reply:
Done.